### PR TITLE
Feature/create shopping cart functionality

### PR DESCRIPTION
**Pull request by fullofheal**
image
image
image
1) added redux store functionality to the shopping cart. Allows adding dishes from restaurant view, dynamically change their amount. 
2) 2 views of shopping cart: 
a) desktop view - no shopping cart icon on top. shopping cart automatically appears as a sidebar when at least 1 item is added and is hidden when cart is empty. 
b) mobile view - shopping cart icon in header. when clicked - whole screen drawer cart appears and can be hidden.
3) fixed paulius' comment from previous sprint ticket - restaurant registration form checkout boxes switched to radio.

### CODE REVIEW

File: src/features/client/pages/RestaurantInspect/index.jsx [hunk 1]
```diff
@@ -73,11 +76,29 @@ function RestaurantInspect() {
...
-        })}
-      </Box>
+      <Flex padding={{ base: 0, sm: '0 10px' }} bg="white" gap="10px">
+        <Box bg="white" paddingTop="20px" width="100%" flex="6">
+          {restaurantInfo.restaurantMenu.map((dealSection, i) => {
+            return (
+              <RestaurantInspectSection
+                key={i}
+                dealSection={dealSection}
+                cartOpened={cartItems.length ? true : false}
```
Comment (skkristis) [inline src/features/client/pages/RestaurantInspect/index.jsx:86]: Move logic to component, this should only pass a value

File: src/features/client/components/RestaurantInspectModal/index.jsx [hunk 2]
```diff
@@ -13,14 +13,24 @@ import {
...
   const [dealCount, setDealCount] = useState(1);
   const itemTotal = (dealCount * deal.recipePrice).toFixed(2);
 
+  const dispatch = useDispatch();
+
   return (
-    <Modal isOpen={isOpen} onClose={onClose}>
+    <Modal
+      isOpen={isOpen}
+      onClose={() => {
```
Comment (skkristis) [inline src/features/client/components/RestaurantInspectModal/index.jsx:29]: Move logic to component, pass function

File: src/features/client/components/RestaurantInspectModal/index.jsx [hunk 3]
```diff
@@ -83,7 +93,11 @@ function RestaurantInspectModal({ isOpen, onClose, deal }) {
                 +
               </Button>
             </Flex>
-            <Button bg="blue.400" color="white">
+            <Button
+              bg="blue.400"
+              color="white"
+              onClick={() => dispatch(add({ ...deal, quantity: dealCount }))}
```
Comment (skkristis) [inline src/features/client/components/RestaurantInspectModal/index.jsx:99]: Move logic to component, pass function

File: src/features/client/components/CartComponent/CartDrawer.jsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          padding="2px 8px"
+          bg="blue.400"
+          rounded="xl"
+          color="white"
+          fontSize="12px"
+          position="absolute"
+          bottom="0"
+          right="0"
+        >
+          {cartItems.length
```
Comment (skkristis) [inline src/features/client/components/CartComponent/CartDrawer.jsx:54]: Move logic to component, pass function/value

File: src/features/client/components/CartComponent/CartSideBar.jsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                        <Heading fontSize="16px">{item.recipeName}</Heading>
+                        <Text fontSize="14px" marginTop="10px" color="blue.400">
+                          €{itemTotal}
+                        </Text>
+                      </Box>
+                    </Center>
+                  </Flex>
+                  <Flex alignItems="center" gap="5px">
+                    <Button
+                      onClick={() =>
```
Comment (skkristis) [inline src/features/client/components/CartComponent/CartSideBar.jsx:72]: Move logic to map variable init space, pass function

File: src/features/client/components/CartComponent/CartSideBar.jsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                        dispatch(changeAmount({ id: item.id, index: -1 }))
+                      }
+                      background="none"
+                    >
+                      -
+                    </Button>
+                    <Text>{item.quantity}</Text>
+                    <Button
+                      onClick={() =>
```
Comment (skkristis) [inline src/features/client/components/CartComponent/CartSideBar.jsx:81]: Move logic to map variable init space, pass function

File: src/features/client/components/CartComponent/CartSideBar.jsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                  {cartItems.length}
+                </Box>
+                Go to checkout
+              </Box>
+              <Text>{orderTotal}</Text>
+            </Button>
+          </Box>
+        </Flex>
+      </Box>
+      <Box></Box>
```
Comment (skkristis) [inline src/features/client/components/CartComponent/CartSideBar.jsx:122]: ?

File: src/features/client/components/CartComponent/CartSideBar.jsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                    >
+                      +
+                    </Button>
+                  </Flex>
+                </Flex>
+              );
+            })}
+          </Box>
+          <Spacer />
+          <Box maxWidth="100%">
```
Comment (skkristis) [inline src/features/client/components/CartComponent/CartSideBar.jsx:94]: Does nothing?

File: src/store/reducers/cartReducer.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                  quantity: dish.quantity + action.payload.quantity,
+                }
+              : dish;
+          }),
+        };
+      }
+      return { ...state, list: [...state.list, action.payload] };
+    },
+    changeAmount: (state, action) => {
+      state.list = state.list
```
Comment (skkristis) [inline src/store/reducers/cartReducer.jsx:34]: afaik antipattern, you dont want to manipulate directly.
const newList = ...

return {
...state, list: newList}

File: src/store/reducers/cartReducer.jsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+              : { ...dish, quantity: dish.quantity - 1 };
+          }
+          return dish;
+        })
+        .filter((dish) => {
+          return dish.quantity > 0;
+        });
+    },
+  },
+  extraReducers: () => {},
```
Comment (skkristis) [inline src/store/reducers/cartReducer.jsx:48]: if its needed, will be added, now just dead code. delete

File: src/store/reducers/cartReducer.jsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+                  ...dish,
+                  quantity: dish.quantity + action.payload.quantity,
+                }
+              : dish;
+          }),
+        };
+      }
+      return { ...state, list: [...state.list, action.payload] };
+    },
+    changeAmount: (state, action) => {
```
Comment (skkristis) [inline src/store/reducers/cartReducer.jsx:33]: not consistent naming. could be decreaseAmount, or whatever you think of

File: src/store/reducers/cartReducer.jsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+const initialState = {
+  list: [],
+};
+
+export const cartSlice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    add: (state, action) => {
```
Comment (skkristis) [inline src/store/reducers/cartReducer.jsx:11]: not consistent naming, also very broad. add what? increaseAmount would be better, or you choice of a more descriptive name

**Review by skkristis — CHANGES_REQUESTED** [review]

File: src/features/client/pages/RestaurantInspect/index.jsx (hunk 1 above)
Comment (fullofheal) [reply src/features/client/pages/RestaurantInspect/index.jsx:86]: fixed

File: src/features/client/components/RestaurantInspectModal/index.jsx (hunk 3 above)
Comment (fullofheal) [reply src/features/client/components/RestaurantInspectModal/index.jsx:99]: fixed

File: src/features/client/components/CartComponent/CartDrawer.jsx (hunk 4 above)
Comment (fullofheal) [reply src/features/client/components/CartComponent/CartDrawer.jsx:54]: fixed

File: src/features/client/components/CartComponent/CartSideBar.jsx (hunk 5 above)
Comment (fullofheal) [reply src/features/client/components/CartComponent/CartSideBar.jsx:72]: fixed

File: src/store/reducers/cartReducer.jsx (hunk 9 above)
Comment (fullofheal) [reply src/store/reducers/cartReducer.jsx:34]: yeah, good point, noticed that myself even. fixed.

**Review by skkristis — APPROVED** [review]

**Pull request merged**